### PR TITLE
Fixed Flatten op

### PR DESCRIPTION
**Pull request by wjj19950828**
# Create A Good Pull Request

> 下面的文字请保留在PR说明的最后面，并在提完PR后，根据实际情况勾选确认以下情况  

Please check the follow step before merging this pull request

- [x] Python code style verification
- [x] Review all the code diff by yourself
- [ ] All models(TensorFLow/Caffe/ONNX/PyTorch) testing passed
- [x] Details about your pull request, releated issues

If this PR add new model support, please update `model_zoo.md` and add model to out test model zoos(@wjj19950828)
- [x] New Model Supported
- [ ] No New Model Supported

## Do follow contributes
1、修复```Flatten```，若获取不到输入的shape，通过shape tensor去解决
2、修复```create_parameter```、```Constant```、```ConstantOfShape```、```Where```：
- ```paddle.full```当value为-inf/inf时需转成string
- Where直接用```paddle.where```替代

3、支持HF Marian模型

### CODE REVIEW

File: x2paddle/op_mapper/onnx2paddle/opset9/opset.py [hunk 1]
```diff
@@ -1627,20 +1632,66 @@ def Flatten(self, node):
...
+                            val_x.name + "_shape_second"
+                        ]
+                    },
+                    outputs=[val_x.name + "_all_shape"],
+                    axis=0)
+                self.paddle_graph.add_layer(
+                    'paddle.reshape',
+                    inputs={'x': val_x.name},
+                    outputs=[node.name],
+                    shape=val_x.name + "_all_shape")
```
Comment (jiangjiajun) [inline x2paddle/op_mapper/onnx2paddle/opset9/opset.py]: 这个没法直接用paddle.flatten吗

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: x2paddle/op_mapper/onnx2paddle/opset9/opset.py (hunk 1 above)
Comment (wjj19950828) [reply x2paddle/op_mapper/onnx2paddle/opset9/opset.py]: ONNX flatten与paddle.flatten定义不一样

ONNX指定axis，输出为2D tensor

Paddle为指定start_axis以及stop_axis，在这个区间变成连续，其他不变

eg:inputs tensor->[1, 2, 3, 4], axis=2

ONNX：outputs->[2, 12]
Paddle: outputs->[2, 3, 4]

Comment (jiangjiajun) [reply x2paddle/op_mapper/onnx2paddle/opset9/opset.py]: 那是否这样会更简洁
```
a = paddle.flatten(x, start_axis=0, stop_axis=axis)
b = paddle.reshape(a, (0, -1))
```

Comment (wjj19950828) [reply x2paddle/op_mapper/onnx2paddle/opset9/opset.py]: Done.

**Review by jiangjiajun — APPROVED** [review]
